test/allocation_tracker-bench.cc

```diff
@@ -39,7 +44,7 @@ void perform_memory_operations(bool track_allocations,
   std::mt19937 gen(rd());
 
   for (auto _ : state) {
-    state.PauseTiming();
+    //    state.PauseTiming();
```

Comment:
why does this need to be commented out?

Reply:
Initially I was measuring the cost of deallocation. In this PR I was optimizing the allocation code path.
I think the correct thing to do is have different benchmarks. I am continuing this work and I'll revisit this.